2.7 Download
================

 | Link: https://bio.liclab.net/scvdb/download

All downloadable data provides users with a centralized access portal aimed at facilitating their access to research resources.
We have integrated the following comprehensive datasets:
(i) scATAC-seq data;
(ii) Fine-mapping results;
(iii) The trait- or disease-relevant score (TRS) of each single-cell generated by g-chromVAR and SCAVENGE methods;
(iv) Results of gene and TF related analysis;
(v) Gene regulation annotation data.

2.7.1 Download trait- or disease-relevant cell score (TRS) data for each sample
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. image:: ../img/download/overview.png

2.7.1.1 Overview of scATAC-seq data: ``txt`` file
*******************************************************************

==== =========================== ==================================================================================================
#    Column name                 Description
==== =========================== ==================================================================================================
1    f_sample_id                 The unique identifier of the single-cell sample, used for database operations.
2    f_gse_id                    GSE ID
3    f_genome                    The reference genome of the single-cell sample.
4    f_geo_id                    GEO ID
5    f_label                     The unique identifier for the single-cell sample, used as the file name during data processing.
6    f_pmid                      PMID
7    f_species                   The species information of the single-cell sample. All data belongs to humans.
8    f_tissue_type               The tissue type of the single-cell sample.
9    f_sequencing_type           The sequencing type of the single-cell sample.
10   f_health_type               The health type of the single-cell sample.
11   f_health_type_description   Detailed information on the health type of the single-cell sample.
12   f_description               Detailed information on the content of the single-cell sample.
13   f_source                    The source name of the single-cell sample.
14   f_source_url                The link to the source of the single-cell sample.
15   f_counts_layer              The layer name of the counts matrix stored in the Seurat object of the single-cell sample.
16   f_sample_exist              The single-cell sample contains multiple sample information.
17   f_cell_count                The number of cells in the single-cell sample.
18   f_cell_type_count           The number of cell types in the single-cell sample.
19   f_index                     The unique index identifier of the single-cell sample has no meaning and is only used for sorting.
==== =========================== ==================================================================================================

.. note::

    When downloading files, some browsers will directly open the ``txt`` file and need to save the file by right-click.

.. image:: ../img/download/txt_download.png


2.7.1.2 scATC-seq data: ``H5AD`` file
*******************************************************************

Read the information of ``sample_id_1``.

.. code-block:: python
    :linenos:

    >>> data
    AnnData object with n_obs × n_vars = 36721 × 414680
        obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score', 'barcode', 'n_genes', 'n_counts', 'cell_type', 'UMAP1', 'UMAP2', 'barcodes'
        var: 'count', 'selected', 'chr', 'start', 'end', 'n_cells'
        uns: 'doublet_rate', 'macs3', 'params', 'project_name', 'project_version', 'reference_sequences', 'scrublet_sim_doublet_score', 'step'
        obsm: 'fragment_paired'
    >>>
    >>>
    >>> data.var
                                count  selected   chr      start        end  n_cells
    index
    chr1:237500-238000          316.0      True  chr1     237500     238000      296
    chr1:238000-238500          316.0      True  chr1     238000     238500      296
    chr1:540500-541000          222.0      True  chr1     540500     541000      217
    chr1:541000-541500          222.0      True  chr1     541000     541500      217
    chr1:713500-714000        10773.0      True  chr1     713500     714000    10145
    ...                           ...       ...   ...        ...        ...      ...
    chrX:155232500-155233000    246.0      True  chrX  155232500  155233000      225
    chrX:155233500-155234000    200.0      True  chrX  155233500  155234000      186
    chrX:155234000-155234500    200.0      True  chrX  155234000  155234500      186
    chrX:155260000-155260500    603.0      True  chrX  155260000  155260500      563
    chrX:155260500-155261000    603.0      True  chrX  155260500  155261000      563

    [414680 rows x 6 columns]
    >>>
    >>>
    >>> data.obs
                        n_fragment  frac_dup  frac_mito       tsse  doublet_probability  doublet_score             barcode  n_genes  n_counts    cell_type      UMAP1      UMAP2            barcodes
    index
    AAACGAAAGAACGACC-1       24764  0.613793        0.0  14.751286             0.102154       0.095522  AAACGAAAGAACGACC-1    46094     49528      Tumor 4  10.567199  -4.781785  AAACGAAAGAACGACC-1
    AAACGAAAGAATACTG-1        2506  0.389822        0.0  14.333112             0.185441       0.001557  AAACGAAAGAATACTG-1     4809      5012      Myeloid   1.443223  13.324852  AAACGAAAGAATACTG-1
    AAACGAAAGACACGGT-1        4923  0.478827        0.0  23.241852             0.124562       0.040230  AAACGAAAGACACGGT-1     9438      9846         Treg  -1.004199  -7.261578  AAACGAAAGACACGGT-1
    AAACGAAAGACCCTAT-1        3674  0.443755        0.0  21.428571             0.172410       0.007480  AAACGAAAGACCCTAT-1     7059      7348            B  -5.697628  13.187097  AAACGAAAGACCCTAT-1
    AAACGAAAGAGGTACC-1        7178  0.488674        0.0  20.920746             0.152831       0.018101  AAACGAAAGAGGTACC-1    13666     14356      CD8 TEx  -5.956334  -3.010488  AAACGAAAGAGGTACC-1
    ...                        ...       ...        ...        ...                  ...            ...                 ...      ...       ...          ...        ...        ...                 ...
    TTTGTGTTCGAGGCTC-1        4853  0.432597        0.0  17.623604             0.179749       0.004054  TTTGTGTTCGAGGCTC-1     9306      9706         Treg   1.477226  -8.637981  TTTGTGTTCGAGGCTC-1
    TTTGTGTTCGGGTCCA-1        5016  0.492256        0.0  24.892704             0.174884       0.006297  TTTGTGTTCGGGTCCA-1     9551     10032         Treg   2.348910  -6.036977  TTTGTGTTCGGGTCCA-1
    TTTGTGTTCGTCCCAT-1       12915  0.498855        0.0  15.457507             0.122509       0.042428  TTTGTGTTCGTCCCAT-1    24172     25830      CD8 TEx  -8.256992  -3.043979  TTTGTGTTCGTCCCAT-1
    TTTGTGTTCTCTTCCT-1        5429  0.461569        0.0  19.229330             0.173898       0.006765  TTTGTGTTCTCTTCCT-1    10422     10858         Treg   2.174267  -8.784227  TTTGTGTTCTCTTCCT-1
    TTTGTGTTCTGCCGAG-1        3275  0.425842        0.0  16.528926             0.151769       0.018755  TTTGTGTTCTGCCGAG-1     6310      6550  Naive CD8 T  -0.882584   1.916430  TTTGTGTTCTGCCGAG-1

    [36721 rows x 13 columns]


2.7.1.3 The result data of method g-ChromVAR: ``H5AD`` file
*******************************************************************

Read the information of ``sample_id_1``.

 | ``obs``: Cell
 | ``var``: Trait or disease
 | ``X``: Z-score

.. code-block:: python
    :linenos:

    >>> data
    AnnData object with n_obs × n_vars = 36721 × 15805
        obs: 'f_sample_id', 'f_barcodes', 'f_cell_type', 'f_sample', 'f_umap_x', 'f_umap_y', 'f_tsse', 'f_index', 'f_cell_type_index'
        var: 'f_trait_id', 'f_trait_code', 'f_source_genome', 'f_trait_abbr', 'f_trait', 'f_variant_count'
    >>>
    >>> data.var
                        f_trait_id                               f_trait_code f_source_genome                 f_trait_abbr                                            f_trait  f_variant_count
    f_trait_id
    trait_id_826      trait_id_826          CAUSALdb_Appendicitis_PE06234_672            hg19         Appendicitis_PE06234                                       Appendicitis               13
    trait_id_2146    trait_id_2146                  CAUSALdb_COE_FG02496_3096            hg19                  COE_FG02496                                Cancer of esophagus                2
    trait_id_3466    trait_id_3466  CAUSALdb_EHKPCAORROACYBNITLY_FG00466_5927            hg19  EHKPCAORROACYBNITLY_FG00466  Ever had known person concerned about, or reco...                1
    trait_id_1156    trait_id_1156                  CAUSALdb_BNT_F900340_4465            hg19                  BNT_F900340                            Benign neoplasm: Testis                1
    trait_id_1816    trait_id_1816                   CAUSALdb_CI_FG00089_4526            hg19                   CI_FG00089                                      Carrot intake               21
    ...                        ...                                        ...             ...                          ...                                                ...              ...
    trait_id_15801  trait_id_15801                            UKBB_Worrier_43            hg19                      Worrier                                            Worrier             5683
    trait_id_15802  trait_id_15802                     UKBB_Worry_Too_Long_85            hg19               Worry_Too_Long                 Worry too long after embarrassment             3225
    trait_id_15803  trait_id_15803                                UKBB_eBMD_6            hg19                         eBMD                Estimated heel bone mineral density            37155
    trait_id_15804  trait_id_15804                               UKBB_eGFR_15            hg19                         eGFR  Estimated glomerular filtration rate (serum cr...            35955
    trait_id_15805  trait_id_15805                             UKBB_eGFRcys_3            hg19                      eGFRcys   Estimated glomerular filtration rate (cystain C)            37319

    [15805 rows x 6 columns]
    >>>
    >>> data.obs
                        f_sample_id          f_barcodes  f_cell_type   f_sample   f_umap_x   f_umap_y     f_tsse  f_index  f_cell_type_index
    index
    AAACGAAAGAACGACC-1  sample_id_1  AAACGAAAGAACGACC-1      Tumor 4  GSE129785  10.567199  -4.781785  14.751286        1                  0
    AAACGAAAGAATACTG-1  sample_id_1  AAACGAAAGAATACTG-1      Myeloid  GSE129785   1.443223  13.324852  14.333112        2                  0
    AAACGAAAGACACGGT-1  sample_id_1  AAACGAAAGACACGGT-1         Treg  GSE129785  -1.004199  -7.261578  23.241852        3                  0
    AAACGAAAGACCCTAT-1  sample_id_1  AAACGAAAGACCCTAT-1            B  GSE129785  -5.697628  13.187097  21.428571        4                  0
    AAACGAAAGAGGTACC-1  sample_id_1  AAACGAAAGAGGTACC-1      CD8 TEx  GSE129785  -5.956334  -3.010488  20.920746        5                  0
    ...                         ...                 ...          ...        ...        ...        ...        ...      ...                ...
    TTTGTGTTCGAGGCTC-1  sample_id_1  TTTGTGTTCGAGGCTC-1         Treg  GSE129785   1.477226  -8.637981  17.623604    36717               4065
    TTTGTGTTCGGGTCCA-1  sample_id_1  TTTGTGTTCGGGTCCA-1         Treg  GSE129785   2.348910  -6.036977  24.892704    36718               4066
    TTTGTGTTCGTCCCAT-1  sample_id_1  TTTGTGTTCGTCCCAT-1      CD8 TEx  GSE129785  -8.256992  -3.043979  15.457507    36719               3897
    TTTGTGTTCTCTTCCT-1  sample_id_1  TTTGTGTTCTCTTCCT-1         Treg  GSE129785   2.174267  -8.784227  19.229330    36720               4067
    TTTGTGTTCTGCCGAG-1  sample_id_1  TTTGTGTTCTGCCGAG-1  Naive CD8 T  GSE129785  -0.882584   1.916430  16.528926    36721               2767

    [36721 rows x 9 columns]
    >>>
    >>> data.X.todense()
    matrix([[ 0.        ,  0.        ,  0.        , ...,  1.34798235,
              0.13897425,  0.46950752],
            [ 0.        ,  0.        ,  0.        , ..., -0.27093183,
             -0.28416698,  0.2759976 ],
            [ 0.        ,  0.        ,  0.        , ..., -0.6249468 ,
              0.11480793, -1.2071487 ],
            ...,
            [ 0.        ,  0.        ,  0.        , ..., -0.40784247,
              0.35490693, -0.85452906],
            [ 0.        ,  0.        ,  0.        , ...,  0.50343663,
              0.07536454,  0.42840868],
            [ 0.        ,  0.        ,  0.        , ..., -0.82765052,
              0.20382107,  0.89792407]])

2.7.1.4 The result data of method SCAVENGE: ``H5AD`` file
*******************************************************************

Read the information of ``sample_id_1``.

 | ``obs``: Cell
 | ``var``: Trait or disease
 | ``X``: TRS

.. code-block:: python
    :linenos:

    >>> data
    AnnData object with n_obs × n_vars = 36721 × 15805
        obs: 'f_sample_id', 'f_barcodes', 'f_cell_type', 'f_sample', 'f_umap_x', 'f_umap_y', 'f_tsse', 'f_index', 'f_cell_type_index'
        var: 'f_trait_id', 'f_trait_code', 'f_source_genome', 'f_trait_abbr', 'f_trait', 'f_variant_count'
    >>>
    >>>
    >>> data.X.todense()
    matrix([[0.        , 0.        , 0.        , ..., 0.11992209, 0.26094234,
             0.35693139],
            [0.        , 0.        , 0.        , ..., 0.50589785, 2.59232072,
             1.68724861],
            [0.        , 0.        , 0.        , ..., 0.10034563, 0.40161146,
             0.31860852],
            ...,
            [0.        , 0.        , 0.        , ..., 0.03006235, 0.37951727,
             0.08840483],
            [0.        , 0.        , 0.        , ..., 0.09616686, 0.52534063,
             0.47852776],
            [0.        , 0.        , 0.        , ..., 0.21577299, 0.47587153,
             0.39203965]])
    >>>



2.7.2 Download fine-mapping result data for each sample
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. image:: ../img/download/trait.png

2.7.2.1 Overview of fine-mapping result data: ``xlsx`` file
*******************************************************************

==== ==================== ============================================================================================================================================================
#    Column name          Description
==== ==================== ============================================================================================================================================================
1    f_trait_id           The unique identifier of the trait used for searching in the database.
2    f_trait_index        The unique identifier of the trait, used for sorting in the database, corresponds one-to-one with 'f_trait_id'.
3    f_trait_code         The unique identifier of the trait, used as the file name for the file processing procedure.
4    f_trait_abbr         The abbreviation form of the trait.
5    f_trait              Detailed information for the trait.
6    f_type               The trait is classified as one of the types of "disease", "drug", "compound", "health", "subject", "treatment", "symptom", "indicator" or "other".
7    f_icd10              ICD-10
8    f_category           Major categories in ICD-10
9    f_sub_category       Subcategories in ICD-10
10   f_three_category     The third category in ICD-10
11   f_source_id          Unique ID of the trait source cohort.
12   f_source_name        Name of the trait source cohort.
13   f_source_genome      Reference genome of trait source cohort. (Reference genome of the trait before LiftOver)
14   f_variant_count      The number of variant in the trait before LiftOver.
15   f_variant_pp_sum     The total PP value of variant in the trait before LiftOver.
16   f_hg19_count         The number of variant in the trait based on hg19 as a background reference genome.
17   f_hg38_count         The number of variant in the trait based on hg38 as a background reference genome.
18   f_hg19_pp_sum        The total PP value of variant in the trait based on hg19 as a background reference genome.
19   f_hg38_pp_sum        The total PP value of variant in the trait based on hg38 as a background reference genome.
20   f_cohort             The cohort for collecting the trait.
21   f_author             The author of the origin of the trait.
22   f_mesh_id            MESH ID
23   f_mesh_term          MESH TERM
24   f_meta_id            META ID
25   f_popu               Experimental population
26   f_pmid               PMID
27   f_n_case             Case size
28   f_n_control          Control size
29   f_sample_size        Sample size
30   f_filter             Each trait is retained, with a value of 1 for all.
31   f_index              The unique index identifier given in the same source cohort has no meaning and is only used to distinguish different traits in the same source cohort.
32   f_url                The link to download the source of each trait.
==== ==================== ============================================================================================================================================================

2.7.2.2 Fine-mapping result data
*******************************************************************

1. ``txt`` file (``Download`` field)

This file was formed through uniform processing after the original download.

==== ==================== ====================================================================================================
#    Column name          Description
==== ==================== ====================================================================================================
1    trait_code           unique identifier of the trait, used as the file name for the file processing procedure
2    chr                  chromosome in the reference genome coordinate of the source cohort
3    position             position of variant in the reference genome coordinate of the source cohort
4    variant              unique variant identifier
5    rsId                 rsID identifier
6    allele1              reference allele in the reference genome coordinate of the source cohort
7    allele2              alternative allele in the reference genome coordinate of the source cohort. (This allele is the effect allele.)
8    maf                  allele frequency of the minor allele in cohort
9    af                   allele frequency of allele2 (alt)
10   beta                 marginal association effect size from linear mixed model/effect size GWAS
11   se                   standard error on marginal association effect size from linear mixed model/standard error GWAS
12   p_value              p-value GWAS
13   chisq                test statistic for marginal association
14   z_score              original z-score
15   pp                   posterior probability of association from fine-mapping (FINEMAP)
16   beta_posterior       posterior expectation of true effect size
17   sd_posterior         posterior standard deviation of true effect size
18   trait_abbr           abbreviation for the trait
19   trait                detailed information for the trait
20   index                Unique index identifiers based on trait or disease variants are meaningless and can be used to identify the uniqueness of variants.
==== ==================== ====================================================================================================

.. note::

    When collecting fine-mapping result data, some data may not include all columns, and a small number of columns may have null values. Of course, the four columns of "chr", "position", "pp", and "trait" are definitely included.

2. ``bed`` file (``Download (LiftOver)`` field)

SCVdb provides variant coordinates under different reference genomes.

==== ==================== ====================================================================================================
#    Column name          Description
==== ==================== ====================================================================================================
1    None                 chromosome in the reference genome coordinate of the source cohort
2    None                 position of variant in the reference genome coordinate of the source cohort
3    None                 position of variant in the reference genome coordinate of the source cohort (Same as the second column)
4    None                 rsID identifier
5    None                 posterior probability of association from fine-mapping (FINEMAP)
6    None                 abbreviation for the trait
7    None                 Unique index identifiers based on trait or disease variants are meaningless and can be used to identify the uniqueness of variants.
==== ==================== ====================================================================================================


2.7.3 Download other data
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. image:: ../img/download/other_data.png

2.7.3.1 Fine-mapping result data: ``tar.gz`` file
*******************************************************************

Here is the complete download for Part ``2.7.2 Download fine-mapping result data for each sample``.

 | 	Fine-mapping result data (source): ``txt`` file (``Download`` field)
 | 	Fine-mapping result data (hg19/hg38): ``bed`` file (``Download (LiftOver)`` field)

2.7.3.2 Differential gene data: ``txt`` file
*******************************************************************

This file contains differential gene data for all cell types of single-cell samples. Of course, it is after passing the threshold.

==== ==================== ====================================================================================================
#    Column name          Description
==== ==================== ====================================================================================================
1    f_sample_id          unique identifier of scATAC-seq sample
2    f_cell_type          cell type
3    f_gene               gene name
4    f_score              score
5    f_adjusted_p_value   adjusted p value
6    f_log2_fold_change   Log2(Fold change)
7    f_p_value            P-value
==== ==================== ====================================================================================================

.. note::

    You need to download the complete data without threshold filtering, and enter the details page of the sample to download the ``H5AD`` file.

Example: `sample_id_1 <https://bio.liclab.net/scvdb/detail?sampleId=sample_id_1>`_

.. image:: ../img/download/difference_gene_h5ad.png

 | ``obs``: gene
 | ``var``: cell type
 | ``X``: score
 | ``layers``: adjusted p value, Log2(Fold change), P-value


.. code-block:: python
    :linenos:

    >>> data
    AnnData object with n_obs × n_vars = 33501 × 20
        obs: 'n_cells'
        var: 'cell_type', 'size'
        uns: 'diff_genes'
        layers: 'adjusted_p_value', 'log2_fold_change', 'p_value'
    >>>
    >>> data.var
                         cell_type  size
    cell_type
    B                            B   404
    CD8 TEx                CD8 TEx  3898
    Effector CD8 T  Effector CD8 T  1153
    Endothelial        Endothelial   562
    Fibroblasts        Fibroblasts  1325
    Memory CD8 T      Memory CD8 T  4965
    Myeloid                Myeloid   732
    NK1                        NK1   418
    NK2                        NK2  1207
    Naive CD4 T        Naive CD4 T  4059
    Naive CD8 T        Naive CD8 T  2768
    Plasma B              Plasma B   335
    Tfh                        Tfh  4138
    Th1                        Th1   338
    Th17                      Th17  1842
    Treg                      Treg  4068
    Tumor 1                Tumor 1   757
    Tumor 2                Tumor 2   875
    Tumor 3                Tumor 3  1687
    Tumor 4                Tumor 4  1190
    >>>
    >>> data.obs
                     n_cells
    AP006222.2           296
    ENSG00000286448      296
    ENSG00000230021    14992
    ENSG00000228327    10389
    LINC01409          10389
    ...                  ...
    TMLHE               4231
    SPRY3               5205
    VAMP7               7748
    IL9R                5738
    ENSG00000270726      395

    [33501 rows x 1 columns]
    >>>
    >>> data.X
    array([[-16.08996773,  16.2977314 ,  -3.94544339, ...,  22.60018349,
             65.58148956,  41.31241226],
           [ -9.23847771,  38.57592773, -28.23983192, ...,  -8.53127384,
             16.334095  ,  46.58874512],
           [ -9.22247505,  38.53868484, -28.31791878, ...,  -8.08869743,
             16.5304184 ,  46.68078613],
           ...,
           [ -0.73027158,  34.58570862,  42.81091309, ..., -33.24862289,
            -56.29743958, -51.4512825 ],
           [ 12.86117554, -13.21335506,  -1.77498877, ..., -29.03244019,
            -39.19504929, -43.00321579],
           [-16.56791496, -32.8029213 ,   2.89613366, ...,  38.49712753,
             32.102005  , -17.40989685]])
    >>>


2.7.3.3 Differential TF data: ``txt`` file
*******************************************************************

This file contains differential TF data for all cell types of single-cell samples. Of course, it is after passing the threshold.

==== ==================== ====================================================================================================
#    Column name          Description
==== ==================== ====================================================================================================
1    f_sample_id          unique identifier of scATAC-seq sample
2    f_cell_type          cell type
3    f_tf                 transcription factor name
4    f_tf_id              rsID identifier
5    f_p_value            P-value
6    f_adjusted_p_value   adjusted p value
7    f_log2_fold_change   Log2(Fold change)
==== ==================== ====================================================================================================

.. note::

    You need to download the complete data without threshold filtering, and enter the details page of the sample to download the ``H5AD`` file.

Example: `sample_id_1 <https://bio.liclab.net/scvdb/detail?sampleId=sample_id_1>`_

.. image:: ../img/download/difference_tf_h5ad.png

 | ``obs``: TF
 | ``var``: cell type
 | ``X``: P-value
 | ``layers``: adjusted p value, Log2(Fold change)


.. code-block:: python
    :linenos:

    >>> data
    AnnData object with n_obs × n_vars = 1165 × 20
        obs: 'id', 'name'
        var: 'cell_type', 'size'
        layers: 'adjusted_p_value', 'log2_fold_change'
    >>>
    >>> data.obs
                                                id        name
    index
    AC023509.3+M02872_2.00  AC023509.3+M02872_2.00  AC023509.3
    AC138696.1+M04597_2.00  AC138696.1+M04597_2.00  AC138696.1
    AHR+M09817_2.00                AHR+M09817_2.00         AHR
    AIRE+M09375_2.00              AIRE+M09375_2.00        AIRE
    ALX1+M05327_2.00              ALX1+M05327_2.00        ALX1
    ...                                        ...         ...
    ZSCAN4+M02919_2.00          ZSCAN4+M02919_2.00      ZSCAN4
    ZSCAN5+M04460_2.00          ZSCAN5+M04460_2.00      ZSCAN5
    ZSCAN5C+M08390_2.00        ZSCAN5C+M08390_2.00     ZSCAN5C
    ZSCAN9+M04466_2.00          ZSCAN9+M04466_2.00      ZSCAN9
    ZZZ3+M01272_2.00              ZZZ3+M01272_2.00        ZZZ3

    [1165 rows x 2 columns]
    >>>
    >>> data.X
    array([[1.01662951e-01, 1.74660328e-01, 2.50931395e-01, ...,
            6.34538848e-02, 7.25013930e-02, 5.10951651e-05],
           [2.07562180e-01, 1.93983057e-01, 2.10357488e-01, ...,
            3.01950908e-01, 3.46950746e-01, 8.56932171e-02],
           [2.40413032e-01, 9.76634287e-02, 6.66147596e-01, ...,
            2.68301581e-01, 1.75328527e-02, 1.26211337e-03],
           ...,
           [4.38363454e-01, 1.43397437e-01, 4.24778841e-01, ...,
            7.15759727e-03, 5.41759614e-02, 9.35845828e-12],
           [4.86767592e-01, 1.47841135e-01, 5.32381338e-01, ...,
            2.74014131e-01, 1.13489445e-05, 6.38005942e-11],
           [1.61418404e-01, 3.23724955e-01, 4.50586827e-02, ...,
            2.66768124e-01, 7.84328678e-02, 4.08885306e-07]])
    >>>

2.7.3.4 The result data of enriched genes for traits or diseases through MAGMA: ``txt`` file
*******************************************************************************************************





2.7.3.1 Gene regulation annotation data: ``txt`` file
*******************************************************************









